dev/NavigationView/NavigationViewItem.cpp

```diff
@@ -54,16 +102,66 @@ void NavigationViewItem::OnApplyTemplate()
         UpdateIsClosedCompact();
     }
 
+    // Retrieve reference to NavigationView
+    if (auto nvImpl = winrt::get_self<NavigationView>(GetNavigationView()))
+    {
+        if (auto repeater = GetTemplateChildT<winrt::ItemsRepeater>(c_repeater, controlProtected))
+        {
+            m_repeater.set(repeater);
+
+            // Primary element setup happens in NavigationView
+            m_repeaterElementPreparedRevoker = repeater.ElementPrepared(winrt::auto_revoke, { nvImpl,  &NavigationView::OnRepeaterElementPrepared });
+            m_repeaterElementClearingRevoker = repeater.ElementClearing(winrt::auto_revoke, { nvImpl, &NavigationView::OnRepeaterElementClearing });
+
+            repeater.ItemTemplate(*(nvImpl->m_navigationViewItemsFactory));
+        }
+
+        UpdateRepeaterItemsSource();
+    }
+
+    if (auto flyoutContentGrid = GetTemplateChildT<winrt::Grid>(c_flyoutContentGrid, controlProtected))
+    {
+        m_flyoutContentGrid.set(flyoutContentGrid);
+    }
+
+    if (auto flyoutTitleContentPresenterGrid = GetTemplateChildT<winrt::Grid>(c_flyoutTitleContentPresenterGrid, controlProtected))
+    {
+        m_flyoutTitleContentPresenterGrid.set(flyoutTitleContentPresenterGrid);
+    }
+
+    if (auto flyoutTitleContentPresenter = GetTemplateChildT<winrt::ContentPresenter>(c_flyoutTitleContentPresenter, controlProtected))
+    {
+        m_flyoutTitleContentPresenter.set(flyoutTitleContentPresenter);
+    }
+
     m_appliedTemplate = true;
+    UpdateItemIndentation();
     UpdateVisualStateNoTransition();
 
     auto visual = winrt::ElementCompositionPreview::GetElementVisual(*this);
     NavigationView::CreateAndAttachHeaderAnimation(visual);
 }
 
+void NavigationViewItem::UpdateRepeaterItemsSource()
+{
+    auto const itemsSource = [this]()
```

Comment:
>itemsSource [](start = 15, length = 11)

nit: move this lambda into the if block where it's result is used so we don't do this when it isn't needed.